### PR TITLE
Release note for 2024-01-08

### PR DESCRIPTION
**Pull request by charleswool**
Initial draft for release note 2024-01-08

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,42 @@
...
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.
+  * Fix for LTS to look at AgentPoolResources.
+  * Fix for a put MC operation with an API version that doesn't have the servicemeshprofile triggers an "invalid mode" error during RP validations.
+
+* Behavioral Change
+  * ManagedCluster, AgentPool are now deprecated fields in agentpool goals.
```
Comment (shashankbarsin) [inline CHANGELOG.md:23]: goals?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
+  * The default ResponseHeaderTimeout is now set to 55s to prevent abnormal slow operation caused by hanging connection.
```
Comment (shashankbarsin) [inline CHANGELOG.md:24]: Needs more context. Response header timeout for?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,42 @@
...
+  * Community images are now added into reduction pipelines for cost saving.
```
Comment (shashankbarsin) [inline CHANGELOG.md:25]: Reduction pipelines? Context? Is this external facing?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,42 @@
...
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-driver) is now increased to 300Mi.
```
Comment (shashankbarsin) [inline CHANGELOG.md:26]: - need to remove 'en-us'. 
- Increased from what value to 300Mi?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,42 @@
...
+  * Expanders flag is removed from AutoscalerProfile.
```
Comment (shashankbarsin) [inline CHANGELOG.md:27]: Context? From which API version? Any next steps for users having this on their existing clusters?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,42 @@
...
+  * Config checksum is added to cns daemonset spec.
```
Comment (shashankbarsin) [inline CHANGELOG.md:28]: Context? Why was this change introduced? To rollover daemonset when ConfigMap is changed?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,42 @@
...
+  * Orchestrator version is moved to new validator removing from server/validation.
```
Comment (shashankbarsin) [inline CHANGELOG.md:29]: Is this external facing?

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,42 @@
...
+
+* Component Updates
+  * Windows Kubernetes CVE fixes for 3P Windows customers, for: [CVE-2023-5528](https://github.com/advisories/GHSA-hq6q-c2x6-hmch)
```
Comment (shashankbarsin) [inline CHANGELOG.md:32]:
```suggestion
  * Windows Kubernetes CVE fixes for [CVE-2023-5528](https://github.com/advisories/GHSA-hq6q-c2x6-hmch)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,42 @@
...
+  * Update ama-logs addon image to 3.1.16
```
Comment (shashankbarsin) [inline CHANGELOG.md:33]: any link to release notes?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,42 @@
...
+  * Bumpe cloud-controller-manager image to [v1.25.23](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.25.23), [v1.26.18](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.26.18), [v1.27.12](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.27.12), [v1.28.4](https://github.com/kubernetes-sigs/cloud-provider-azure/releases/tag/v1.28.4).
+  * Bump agentbaker version to v0.20231122.3, v0.20231206.0.
```
Comment (shashankbarsin) [inline CHANGELOG.md:35]: We can remove this line and we just need to cover the vhd notes

Comment (charleswool) [reply CHANGELOG.md:35]: Updated

File: CHANGELOG.md (hunk 9 above)
Comment (charleswool) [reply CHANGELOG.md:33]: updated

File: CHANGELOG.md (hunk 7 above)
Comment (charleswool) [reply CHANGELOG.md:29]: Removed

File: CHANGELOG.md (hunk 6 above)
Comment (charleswool) [reply CHANGELOG.md:28]: Updated
  * Config checksum is added to cns daemonset spec will roll CNS pods that need to pick up new config values in azure-cns-configmap, this is to resolve the recent Windows goroutine leak.

File: CHANGELOG.md (hunk 5 above)
Comment (charleswool) [reply CHANGELOG.md:27]: Updated

File: CHANGELOG.md (hunk 1 above)
Comment (charleswool) [reply CHANGELOG.md]: Updated

  * ManagedCluster, AgentPool are now deprecated fields in agentpool goals. As goal should be including all necessary information. It is genrated from ManagedCluster abnd AgentPool and they should no longer be used.

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,39 @@
...
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) is now increased to 300Mi.
+  * Expanders flag is removed from AutoscalerProfile from November API
+  * Config checksum is added to cns daemonset spec will roll CNS pods that need to pick up new config values in azure-cns-configmap, this is to resolve the recent Windows goroutine leak.
+    
+* Component Updates
+  * Windows Kubernetes CVE fixes for [CVE-2023-5528](https://github.com/advisories/GHSA-hq6q-c2x6-hmch)
+  * Update ama-logs addon image to [3.1.16](https://github.com/microsoft/Docker-Provider/blob/ci_prod/ReleaseNotes.md)
+  * Azure Windows 2019 Image has been updated to [Azure Windows - 17763.5206.231213](https://github.com/Azure/AKS/blob/2024-01-08/vhd-notes/AKSWindows/2019/17763.5206.231213.txt)
+  * Azure Windows 2022 Image has been updated to [Azure Windows - 20348.2159.231213](https://github.com/Azure/AKS/blob/2024-01-08/vhd-notes/AKSWindows/2022/20348.2159.231213.txt)
```
Comment (allyford) [inline CHANGELOG.md:34]:
```suggestion
  * Windows Server 2022 Image has been updated to [20348.2159.231213](https://github.com/Azure/AKS/blob/2024-01-08/vhd-notes/AKSWindows/2022/20348.2159.231213.txt)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,39 @@
...
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.
+  * Fix for LTS to look at AgentPoolResources.
+  * Fix for a put MC operation with an API version that doesn't have the servicemeshprofile triggers an "invalid mode" error during RP validations.
+  * Fix for the wrong MCR URL for Keda in AGC caused by cloud Environment value not being passed to the addon chart.
+
+* Behavioral Change
```
Comment (allyford) [inline CHANGELOG.md:23]: [Artifact Streaming](https://learn.microsoft.com/azure/aks/artifact-streaming) can now be enabled on an existing nodepool.

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,39 @@
...
+* Starting Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
+* Staring with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
+* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024.
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.
```
Comment (allyford) [inline CHANGELOG.md:18]: I'm checking with eng on this one to see if we want to reword it

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,39 @@
...
+* Starting Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
+* Staring with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
+* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024.
```
Comment (allyford) [inline CHANGELOG.md:13]:
```suggestion
* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024. See [release calendar](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar:~:text=patched%20and%20supported.-,AKS%20Kubernetes%20release%20calendar,-View%20the%20upcoming).
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,39 @@
...
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.
+  * Fix for LTS to look at AgentPoolResources.
```
Comment (allyford) [inline CHANGELOG.md:19]: Link to LTS docs? And how does that impact customers?

**Comment by allyford** [comment]
Make sure to add links wherever possible! @charleswool

File: CHANGELOG.md (hunk 13 above)
Comment (allyford) [reply CHANGELOG.md:18]: Update: Remove this
```suggestion
```

File: CHANGELOG.md (hunk 12 above)
Comment (kaarthis) [reply CHANGELOG.md:23]: Wherever possible please add GH work items translating to the bug .

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,39 @@
...
+  * Fix for a put MC operation with an API version that doesn't have the servicemeshprofile triggers an "invalid mode" error during RP validations.
+  * Fix for the wrong MCR URL for Keda in AGC caused by cloud Environment value not being passed to the addon chart.
+
+* Behavioral Change
+  * ManagedCluster, AgentPool are now deprecated fields in agentpool goals. As goal should be including all necessary information and is generated from ManagedCluster abnd AgentPool. Thus, they should no longer be used.
+  * The default ResponseHeaderTimeout (which specifies the amount of time to wait for a server's response headers after fully writing the request, does not include the time to read the response body) is now set to 55s to prevent abnormal slow operation caused by hanging connection.
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) is now increased to 300Mi.
+  * Expanders flag is removed from AutoscalerProfile from November API
```
Comment (kaarthis) [inline CHANGELOG.md:27]: Perhaps some doc reference to autoscalerprofile would help.

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,5 +1,39 @@
...
+  * Config checksum is added to cns daemonset spec will roll CNS pods that need to pick up new config values in azure-cns-configmap, this is to resolve the recent Windows goroutine leak.
+    
+* Component Updates
+  * Windows Kubernetes CVE fixes for [CVE-2023-5528](https://github.com/advisories/GHSA-hq6q-c2x6-hmch)
+  * Update ama-logs addon image to [3.1.16](https://github.com/microsoft/Docker-Provider/blob/ci_prod/ReleaseNotes.md)
+  * Azure Windows 2019 Image has been updated to [Azure Windows - 17763.5206.231213](https://github.com/Azure/AKS/blob/2024-01-08/vhd-notes/AKSWindows/2019/17763.5206.231213.txt)
```
Comment (kaarthis) [inline CHANGELOG.md:33]: Why is AZ linux (mariner) VHD missing here ?https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/release-notes/AKSCBLMarinerV2/gen2/202401.03.0.txt

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,5 +1,39 @@
...
+### Release notes
+
+* Bug Fixes 
+  * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.
+  * Fix for LTS to look at AgentPoolResources.
+  * Fix for a put MC operation with an API version that doesn't have the servicemeshprofile triggers an "invalid mode" error during RP validations.
+  * Fix for the wrong MCR URL for Keda in AGC caused by cloud Environment value not being passed to the addon chart.
+
+* Behavioral Change
+  * ManagedCluster, AgentPool are now deprecated fields in agentpool goals. As goal should be including all necessary information and is generated from ManagedCluster abnd AgentPool. Thus, they should no longer be used.
```
Comment (kaarthis) [inline CHANGELOG.md:24]: Any doc link talking about this?

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,5 +1,39 @@
 # Azure Kubernetes Service Changelog
 
+## Release 2024-01-08
```
Comment (bridgetkromhout) [inline CHANGELOG.md:3]: Do we want to update this to whatever date this is published?

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,5 +1,39 @@
...
+* Starting January 2024, due to Gatekeeper Upstream removing validation for constraint template contents at create/update time, [the Azure Policy Add-On](https://learn.microsoft.com/azure/governance/policy/concepts/policy-for-kubernetes#install-azure-policy-add-on-for-aks:~:text=exception%20YAML.-,Install%20Azure%20Policy%20Add%2Don%20for%20AKS,-Before%20you%20install) will now no longer support this. The Azure Policy Add-On will report [‘InvalidConstraint/Template’ compliance reason code](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons) for detected errors after constraint template admission. This change does not impact [other compliance reason codes](https://learn.microsoft.com/azure/governance/policy/how-to/determine-non-compliance#aks-resource-provider-mode-compliance-reasons). Customers are encouraged to continue to follow best practices when updating Azure Policy for Kubernetes definitions (i.e. [Gator CLI](https://open-policy-agent.github.io/gatekeeper/website/docs/gator/).
```
Comment (bridgetkromhout) [inline CHANGELOG.md:10]: > will now no longer support this.

Do we want to be more explicit and spell out "this" so people don't need to try to figure out what exactly applies?

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,5 +1,39 @@
...
+* Staring with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
+* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024.
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.
+  * Fix for LTS to look at AgentPoolResources.
+  * Fix for a put MC operation with an API version that doesn't have the servicemeshprofile triggers an "invalid mode" error during RP validations.
```
Comment (shashankbarsin) [inline CHANGELOG.md:20]:
```suggestion
  * PUT managedCluster operations on API versions that didn't support serviceMeshProfile resulted in "invalid mode" error response to the API requests. This issue has now been fixed.
```

File: CHANGELOG.md (hunk 1 above)
Comment (shashankbarsin) [reply CHANGELOG.md]: From a customer's perspective, what goals are we referring to? something related to API?

File: CHANGELOG.md (hunk 2 above)
Comment (shashankbarsin) [inline CHANGELOG.md]: Sharing questions on this in internal thread

File: CHANGELOG.md (hunk 18 above)
Comment (shashankbarsin) [reply CHANGELOG.md:24]: from a customer's perspective, what are we referring to with respect to 'goals'? something related to API?

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,5 +1,39 @@
...
+  * Fix for the wrong MCR URL for Keda in AGC caused by cloud Environment value not being passed to the addon chart.
+
+* Behavioral Change
+  * ManagedCluster, AgentPool are now deprecated fields in agentpool goals. As goal should be including all necessary information and is generated from ManagedCluster abnd AgentPool. Thus, they should no longer be used.
+  * The default ResponseHeaderTimeout (which specifies the amount of time to wait for a server's response headers after fully writing the request, does not include the time to read the response body) is now set to 55s to prevent abnormal slow operation caused by hanging connection.
```
Comment (shashankbarsin) [inline CHANGELOG.md:25]: Sharing questions on this in internal thread

File: CHANGELOG.md [hunk 23]
```diff
@@ -1,5 +1,39 @@
...
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) is now increased to 300Mi.
```
Comment (shashankbarsin) [inline CHANGELOG.md:26]: increased from?

File: CHANGELOG.md (hunk 16 above)
Comment (shashankbarsin) [reply CHANGELOG.md:27]: - what impact or behavior change does this removal have? does this impact customers with existing clusters in any way?
- for API version, we need to mention the official format. for example, 2023-11-01, 2023-11-01-preview,...

File: CHANGELOG.md (hunk 20 above)
Comment (charleswool) [reply CHANGELOG.md:10]: Updated to replace with 'validation for constraint template'

File: CHANGELOG.md [hunk 24]
```diff
@@ -1,5 +1,39 @@
...
+* Starting Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
```
[the remark this answers is not in the document]
Comment (charleswool) [reply CHANGELOG.md:11]: Updated

File: CHANGELOG.md [hunk 25]
```diff
@@ -1,5 +1,39 @@
...
+* Staring with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
```
[the remark this answers is not in the document]
Comment (charleswool) [reply CHANGELOG.md:12]: Updated, MB

File: CHANGELOG.md (hunk 14 above)
Comment (charleswool) [reply CHANGELOG.md:13]: Updated

File: CHANGELOG.md (hunk 13 above)
Comment (charleswool) [reply CHANGELOG.md:18]: Updated, thanks for verifying

File: CHANGELOG.md (hunk 15 above)
Comment (charleswool) [reply CHANGELOG.md:19]: Updated

File: CHANGELOG.md (hunk 21 above)
Comment (charleswool) [reply CHANGELOG.md:20]: Updated, thanks for rewording

File: CHANGELOG.md (hunk 12 above)
Comment (charleswool) [reply CHANGELOG.md:23]: Added

File: CHANGELOG.md (hunk 4 above)
Comment (charleswool) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (charleswool) [reply CHANGELOG.md:33]: updated

File: CHANGELOG.md (hunk 11 above)
Comment (charleswool) [reply CHANGELOG.md:34]: Updated

File: CHANGELOG.md (hunk 17 above)
Comment (charleswool) [reply CHANGELOG.md:33]: Thought we should be watching for AzureLinux, added and updated

File: CHANGELOG.md (hunk 16 above)
Comment (charleswool) [reply CHANGELOG.md:27]: It is basically a going back to the old behavior and removing the field we added. It's breaking but it's in preview only so it's OK according to Matthew.

File: CHANGELOG.md (hunk 18 above)
Comment (charleswool) [reply CHANGELOG.md:24]: Confirmed with Hao Yuan, this change should be considered as internal change and no customer awared impact would be caused, Removed

File: CHANGELOG.md (hunk 1 above)
Comment (charleswool) [reply CHANGELOG.md]: Confirmed with Hao Yuan, this one should be removed as it is a internal only change

File: CHANGELOG.md [hunk 26]
```diff
@@ -1,5 +1,39 @@
...
+* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024.
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for artifact streaming update related panic when nil and switches testing to use params/test package functions.
+  * Fix for LTS to look at AgentPoolResources.
+  * Fix for a put MC operation with an API version that doesn't have the servicemeshprofile triggers an "invalid mode" error during RP validations.
+  * Fix for the wrong MCR URL for Keda in AGC caused by cloud Environment value not being passed to the addon chart.
```
[the remark this answers is not in the document]
Comment (charleswool) [reply CHANGELOG.md]: Updated, we didn't see any keda usage at any cloud environment except public cloud before. Thus, theoretically the wrong image URL will cause fail enablement.

File: CHANGELOG.md (hunk 22 above)
Comment (charleswool) [reply CHANGELOG.md:24]: Kindly see the reply in the thread

File: CHANGELOG.md [hunk 27]
```diff
@@ -1,5 +1,39 @@
...
+### Release notes
+
+* Bug Fixes 
+  * Fix for [LTS](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#long-term-support-lts) to look at AgentPoolResources.
+  * PUT managedCluster operations on API versions that didn't support serviceMeshProfile resulted in "invalid mode" error response to the API requests. This issue has now been fixed.
+  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used, which may fail the feature enabling This issue has now been fixed.
+
+* Behavioral Change
+  * [Artifact Streaming](https://learn.microsoft.com/en-us/azure/aks/artifact-streaming) can now be enabled on an existing nodepool.
+  * The default ResponseHeaderTimeout (The timeout of http client used by AKS RP, which specifies the amount of time to wait for a server's response headers after fully writing the request, does not include the time to read the response body) is now set to 55s to prevent abnormal slow operation caused by hanging connection.
```
Comment (tdihp) [inline CHANGELOG.md:24]: What's the "before" behavior for reference of the audience?

File: CHANGELOG.md [hunk 28]
```diff
@@ -1,5 +1,39 @@
...
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) is now increased from 200 Mi to 300Mi.
+  * Expanders flag is removed from [AutoscalerProfile](https://learn.microsoft.com/en-us/azure/aks/hybrid/work-with-autoscaler-profiles#profile-settings) from 2023-11-01-preview since it may cause confusion with existing Expanders. 
+  * Config checksum is added to cns daemonset spec will roll CNS pods that need to pick up new config values in azure-cns-configmap, this is to resolve the recent Windows goroutine leak.
```
Comment (tdihp) [inline CHANGELOG.md:27]: Which feature it is for?

File: CHANGELOG.md [hunk 29]
```diff
@@ -1,5 +1,39 @@
...
+  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used, which may fail the feature enabling This issue has now been fixed.
+
+* Behavioral Change
+  * [Artifact Streaming](https://learn.microsoft.com/en-us/azure/aks/artifact-streaming) can now be enabled on an existing nodepool.
+  * The default ResponseHeaderTimeout (The timeout of http client used by AKS RP, which specifies the amount of time to wait for a server's response headers after fully writing the request, does not include the time to read the response body) is now set to 55s to prevent abnormal slow operation caused by hanging connection.
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) is now increased from 200 Mi to 300Mi.
+  * Expanders flag is removed from [AutoscalerProfile](https://learn.microsoft.com/en-us/azure/aks/hybrid/work-with-autoscaler-profiles#profile-settings) from 2023-11-01-preview since it may cause confusion with existing Expanders.
```
Comment (tdihp) [inline CHANGELOG.md:26]: How does this impact existing users that already uses Expander?

File: CHANGELOG.md (hunk 27 above)
Comment (charleswool) [reply CHANGELOG.md:24]: There wasn't a ResponseHeaderTimeout set before, thus this is a new behavior

File: CHANGELOG.md (hunk 28 above)
Comment (charleswool) [reply CHANGELOG.md:27]: No sensible customer impact, thus removed

File: CHANGELOG.md [hunk 30]
```diff
@@ -1,5 +1,38 @@
...
+  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used, which may fail the feature enabling This issue has now been fixed.
+
+* Behavioral Change
+  * [Artifact Streaming](https://learn.microsoft.com/en-us/azure/aks/artifact-streaming) can now be enabled on an existing nodepool.
+  * The default ResponseHeaderTimeout (The timeout of http client used by AKS RP, which specifies the amount of time to wait for a server's response headers after fully writing the request, does not include the time to read the response body) is now set to 55s to prevent abnormal slow operation caused by hanging connection.
+  * The memory limit for [Azure Key Vault provider for Secrets Store CSI Driver](https://learn.microsoft.com/azure/aks/csi-secrets-store-driver) is now increased from 200 Mi to 300Mi.
+  * [Expanders flag is removed](https://github.com/Azure/azure-rest-api-specs/pull/26440?expand=1&template=data_plane_template.md) from [AutoscalerProfile](https://learn.microsoft.com/en-us/azure/aks/hybrid/work-with-autoscaler-profiles#profile-settings) from 2023-11-01-preview since it may cause confusion with existing Expanders.
```
Comment (shashankbarsin) [inline CHANGELOG.md:26]: en-us needs to be removed

File: CHANGELOG.md [hunk 31]
```diff
@@ -1,5 +1,38 @@
...
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for [LTS](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#long-term-support-lts) to look at AgentPoolResources.
+  * PUT managedCluster operations on API versions that didn't support serviceMeshProfile resulted in "invalid mode" error response to the API requests. This issue has now been fixed.
+  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used, which may fail the feature enabling This issue has now been fixed.
+
+* Behavioral Change
+  * [Artifact Streaming](https://learn.microsoft.com/en-us/azure/aks/artifact-streaming) can now be enabled on an existing nodepool.
```
Comment (shashankbarsin) [inline CHANGELOG.md:23]:
```suggestion
  * [Artifact Streaming](https://learn.microsoft.com/azure/aks/artifact-streaming) can now be enabled on an existing node pool.
```

File: CHANGELOG.md [hunk 32]
```diff
@@ -1,5 +1,38 @@
...
+* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024. See [release calendar](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar:~:text=patched%20and%20supported.-,AKS%20Kubernetes%20release%20calendar,-View%20the%20upcoming).
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for [LTS](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#long-term-support-lts) to look at AgentPoolResources.
+  * PUT managedCluster operations on API versions that didn't support serviceMeshProfile resulted in "invalid mode" error response to the API requests. This issue has now been fixed.
+  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used, which may fail the feature enabling This issue has now been fixed.
```
Comment (shashankbarsin) [inline CHANGELOG.md:20]:
```suggestion
  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used resulting in potential failures in enabling the KEDA addon. This issue has now been fixed.
```

File: CHANGELOG.md [hunk 33]
```diff
@@ -1,5 +1,38 @@
...
+* Starting with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
+* Changes to reduce the kube-reserved memory reservation and eviction threshold will not be available in 1.28 as previously shared due to a release issue. These optimizations will be releasing with AKS Kubernetes minor version 1.29, which previews in January 2024. See [release calendar](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar:~:text=patched%20and%20supported.-,AKS%20Kubernetes%20release%20calendar,-View%20the%20upcoming).
+ 
+### Release notes
+
+* Bug Fixes 
+  * Fix for [LTS](https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#long-term-support-lts) to look at AgentPoolResources.
```
Comment (shashankbarsin) [inline CHANGELOG.md:18]: - need to remove 'en-us'
- needs more context for what/where customer is looking for with respect to agent pool and LTS

File: CHANGELOG.md (hunk 30 above)
Comment (charleswool) [reply CHANGELOG.md:26]: Updated

File: CHANGELOG.md (hunk 33 above)
Comment (charleswool) [reply CHANGELOG.md:18]: Confirmed customer won't notice such change, removed

File: CHANGELOG.md (hunk 29 above)
Comment (charleswool) [reply CHANGELOG.md]: It's going to be a breaking change, it is still in preview though

File: CHANGELOG.md [hunk 34]
```diff
@@ -1,5 +1,37 @@
...
+* Starting with Kubernetes 1.29, the default cgroups implementation on Azure Linux AKS nodes will be cgroupsv2. Older versions of Java, .NET and NodeJS do not support memory querying v2 memory constraints and this will lead to out of memory (OOM) issues for workloads. Please test your applications for cgroupsv2 compliance, and read the [FAQ](https://learn.microsoft.com/troubleshoot/azure/azure-kubernetes/aks-increased-memory-usage-cgroup-v2) for cgroupsv2.
+* Starting with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
```
Comment (wangyira) [inline CHANGELOG.md:13]: should this go under behavior change?

**Review by wangyira — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 34 above)
Comment (wangyira) [reply CHANGELOG.md:13]: "* Starting with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads."

Comment (charleswool) [reply CHANGELOG.md:13]: This one was inherited from the previously announcement. Though I do agree it fits behavior change more, updated

File: CHANGELOG.md (hunk 29 above)
Comment (tdihp) [reply CHANGELOG.md]: @charleswoll ack. Maybe you want to add it doesn't affect current usage of "expander" API call to avoid panic like I had?

**Comment by qpetraroia** [comment]
LGTM

**Review by qpetraroia — APPROVED** [review]

**Review by kaarthis — APPROVED** [review]
Approved

**Review by wangyira — APPROVED** [review]

File: CHANGELOG.md [hunk 35]
```diff
@@ -1,5 +1,40 @@
...
+* Preview Features
+   *  Starting January 2024, the [AI toolchain operator](https://learn.microsoft.com/azure/aks/ai-toolchain-operator) for AKS can now be enabled to set up the GPU infrastructure and deploy OSS models from the Azure CLI in a matter of minutes.
+   *  [Artifact Streaming](https://learn.microsoft.com/azure/aks/artifact-streaming) can now be enabled on an existing node pool.
+  
+* Bug Fixes 
+  * PUT managedCluster operations on API versions that didn't support serviceMeshProfile resulted in "invalid mode" error response to the API requests. This issue has now been fixed.
+  * A wrong MCR URL for [KEDA](https://learn.microsoft.com/azure/aks/keda-about) image in Air Gapped Cloud was previously used resulting in potential failures in enabling the KEDA addon. This issue has now been fixed.
+
+* Behavioral Change
+  * Starting with the `2024-01-01` and `2024-01-02-preview` APIs, we will begin to reject unknown fields in the request payloads.
```
Comment (matthchr) [inline CHANGELOG.md:26]: Consider linking to this issue I just filed which also talks about this: #4060

**Pull request merged**